lib/chef-dk/policyfile_lock.rb

```diff
@@ -264,6 +278,65 @@ def ensure_cache_dir_exists
 
     private
 
+    # Generates a canonical JSON representation of the attributes. Based on
+    # http://wiki.laptop.org/go/Canonical_JSON but not quite as strict, yet.
+    #
+    # In particular:
+    # - String encoding stuff isn't normalized
+    # - We plan to allow floats that fit within the range/precision
+    #   requirements of IEEE 754-2008 binary64 (double precision) numbers
+    # - +/- Infinity and NaN are banned, but float/numeric size aren't checked.
+    #   numerics should be in range [-(2**53)+1, (2**53)-1] to comply with
+    #   IEEE 754-2008
+    #
+    # Recursive, so absurd nesting levels could cause a SystemError. Invalid
+    # input will cause an InvalidPolicyfileAttribute exception.
+    def canonicalize(attributes)
+      unless attributes.kind_of?(Hash)
+        raise "Top level attributes must be a Hash (you gave: #{attributes})"
+      end
+      canonicalize_elements(attributes)
+    end
+
+    def canonicalize_elements(item)
+      case item
+      when Hash
+        elements = item.keys.sort.map do |key|
+          validate_attr_key(key)
+          k = '"' << key.to_s << '":'
+          v = canonicalize_elements(item[key])
+          k << v
+        end
+        "{" << elements.join(',') << "}"
+      when String
+        '"' << item << '"'
```

Comment:
`item.encode('utf-8')` should get the behavior you want and raise if it cannot transcode to utf-8